### ISSUE TITLE
Text search for all code and report objects

### ISSUE THREAD

**Issue by brianpsearle** [DaxStudio/DaxStudio#418]
I'd like to be able to search through all DAX code and report objects for a given string, in one search window with a list showing results of each hit. (similar to Visual Studio Find in files). This would be really helpful for impact analysis when trying to work out how to make changes. For example, doing cleanup and removing unused measures. You could search for the measure name in all the report objects and DAX code. Or, wanting to know where a measure is used in reports or other DAX measures so you can work out whether you can modify it or have to create a slighly different version of it.

**Comment by dgosbell** [comment]
Thanks for suggesting this. While it's definitely worthwhile functionality to have, it's probably beyond the scope of something that is reasonably possible at the moment. 

We discussed this in more depth previously on issue #359 but basically rather than a tool which connects to a single data model like DAX Studio you would need something that could connect and scan at the Power BI tenant level. There is also a lack of supported  APIs from Microsoft in order to be able to track objects used in reports. So at the current point in time this is not something we would plan to add to our road map even though something like this would be very handy to have.

**Comment by brianpsearle** [comment]
Good point, I wasn't thinking beyond the pbix file. #359 was very helpful. Now you mention it, I'd rather not think beyond the pbix file, scary stuff....
